kurtosis-devnet/pkg/kurtosis/sources/deployer/wallets.go

```diff
@@ -67,3 +69,20 @@ func (d *Deployer) getL1ValidatorWallets(deployerArtifact *ktfs.Artifact) ([]*Wa
 
 	return knownWallets, nil
 }
+
+func (d *Deployer) getL1ChainID(genesisArtifact *ktfs.Artifact) (string, error) {
```

Comment:
This looks a bit strange actually - that the extraction need a file writer. Shouldn't  it need a reader? We could make the types more generic for ExtractFiles if needed

Reply:
Agreed it's a bit confusing, you typically don't refer to reading a file into a buffer as a write operation w.r.t. the buffer. Rather than make changes to the underlying ktfs lib, which might be beyond the scope of this PR, I can move this logic into well-named helper functions which make it clear what's happening here.